src/python/pants/core/util_rules/subprocess_environment.py

```diff
@@ -80,27 +77,38 @@ def register_options(cls, register):
         )
 
     @property
-    def env_vars_to_pass_to_subprocesses(self) -> List[str]:
-        ret: List[str] = list(self.options.env_vars)
+    def env_vars_to_pass_to_subprocesses(self) -> Tuple[str, ...]:
+        ret = set(self.options.env_vars)
 
-        # Inject values from the old --lang, --lc-all options if set.
+        # Inject values from the old --lang, --lc-all options if explicitly set, rather than the
+        # default.
         env_var_names = {env_var.split("=", 1)[0] for env_var in self.options.env_vars}
 
         def handle_deprecated(env_var_name: str):
-            val = self.options[env_var_name.lower()]
-            if env_var_name not in env_var_names and val:
-                logger.warning(
-                    f"Setting {env_var_name}={val} on subprocesses due to use of deprecated option "
-                    f"[{self.options_scope}].{env_var_name.lower()}. If you wish to continue to "
-                    f"set this option, do so by adding '{env_var_name}=value' (or '{env_var_name}' "
-                    f"to take the value from Pants's environment) to the "
-                    f"[{self.options_scope}].env_var option."
```

Comment:
We only set the value now if it was explicitly configured. In that case, our normal deprecation message will trigger. So, there is no need for this logging.